mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -306,12 +307,20 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    fit(dataset, handlePersistence)
+  }
+
+  @Since("2.2.0")
+  protected def fit(dataset: Dataset[_], handlePersistence: Boolean): KMeansModel = {
```

Comment:
Hi, sorry for not seeing this earlier.  We should not make a new public method for this.  @ZakariaHili would you mind sending a follow-up PR (under the same JIRA issue) which eliminates this new fit() method?  You should be able to delete lines 311 - 315 here and keep everything else the same.  Thank you!